syntax = "proto3";

package dwarf;

// Kalibrierung starten
message ReqStartCalibration {
}

// Kalibrierung stoppen
message ReqStopCalibration {
}

// GOTO Deep-Sky-Objekt
message ReqGotoDSO {
  double ra = 1;            // Rektaszension
  double dec = 2;           // Deklination
  string target_name = 3;   // Zielname
}

// GOTO Sonnensystem
message ReqGotoSolarSystem {
  int32 index = 1;          // Ziel-Index (1-9)
  double lon = 2;           // GPS-Längengrad
  double lat = 3;           // GPS-Breitengrad
  string target_name = 4;   // Zielname
}

// GOTO stoppen
message ReqStopGoto {
}

// Sonne/Mond Tracking
message ReqTrackSpecialTarget {
  int32 index = 1;  // 0: Sonne, 1: Mond
  double lon = 2;   // GPS-Längengrad
  double lat = 3;   // GPS-Breitengrad
}

// Tracking stoppen
message ReqStopTrackSpecialTarget {
}

// Stacking starten
message ReqCaptureRawLiveStacking {
}

// Stacking stoppen
message ReqStopCaptureRawLiveStacking {
}

// Weitwinkel-Stacking starten
message ReqCaptureWideRawLiveStacking {
}

// Weitwinkel-Stacking stoppen
message ReqStopCaptureWideRawLiveStacking {
}

// Darkframe-Status prüfen
message ReqCheckDarkFrame {
}

message ResCheckDarkFrame {
  int32 progress = 1;  // Fortschritt (0-10000, 2 Dezimalstellen)
  int32 code = 2;      // Fehlercode
}

// Darkframe aufnehmen
message ReqCaptureDarkFrame {
  int32 reshoot = 1;  // 0: Nein, 1: Neu aufnehmen
}

// Darkframe stoppen
message ReqStopCaptureDarkFrame {
}

// Darkframe mit Parametern
message ReqCaptureDarkFrameWithParam {
  int32 exp_index = 1;   // Belichtungs-Index
  int32 gain_index = 2;  // Verstärkungs-Index
  int32 bin_index = 3;   // Binning-Index
  int32 cap_size = 4;    // Anzahl Aufnahmen
}

// Darkframe-Liste abrufen
message ReqGetDarkFrameList {
}

message ResGetDarkFrameInfo {
  int32 exp_index = 1;
  int32 gain_index = 2;
  int32 bin_index = 3;
}

message ResGetDarkFrameInfoList {
  int32 code = 1;
  repeated ResGetDarkFrameInfo results = 2;
}

// Darkframe löschen
message ReqDelDarkFrame {
  int32 exp_index = 1;
  int32 gain_index = 2;
  int32 bin_index = 3;
}

message ReqDelDarkFrameList {
  repeated ReqDelDarkFrame dark_list = 1;
}

message ResDelDarkFrameList {
  int32 code = 1;
}

// GO LIVE
message ReqGoLive {
}

// Ein-Klick GOTO DSO
message ReqOneClickGotoDSO {
  double ra = 1;
  double dec = 2;
  string target_name = 3;
}

// Ein-Klick GOTO Solar
message ReqOneClickGotoSolarSystem {
  int32 index = 1;
  double lon = 2;
  double lat = 3;
  string target_name = 4;
}

// Ein-Klick GOTO Response
message ResOneClickGoto {
  int32 step = 1;      // Schritt-Nummer (10, 20, 30, 40)
  int32 code = 2;      // Fehlercode
  bool all_end = 3;    // Alles abgeschlossen
}

// Ein-Klick GOTO stoppen
message ReqStopOneClickGoto {
}

// EQ-Verifizierung starten
message ReqStartEqSolving {
  double lon = 1;  // GPS-Längengrad
  double lat = 2;  // GPS-Breitengrad
}

message ResStartEqSolving {
  double azi_err = 1;  // Azimut-Fehler (+ = im Uhrzeigersinn)
  double alt_err = 2;  // Höhen-Fehler (+ = nach oben)
  int32 code = 3;
}

// EQ-Verifizierung stoppen
message ReqStopEqSolving {
}
